// precompile-intl-runtime/dist/commonjs/stores/dictionary.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.$locales = exports.$dictionary = exports.addMessages = exports.getClosestAvailableLocale = exports.getMessageFromDictionary = exports.hasLocaleDictionary = exports.getDictionary = exports.getLocaleDictionary = void 0;
// @ts-ignore
const store_1 = require("svelte/store");
const utils_js_1 = require("../includes/utils.js");
let dictionary;
const $dictionary = (0, store_1.writable)({});
exports.$dictionary = $dictionary;
function getLocaleDictionary(locale) {
    return dictionary[locale] || null;
}
exports.getLocaleDictionary = getLocaleDictionary;
function getDictionary() {
    return dictionary;
}
exports.getDictionary = getDictionary;
function hasLocaleDictionary(locale) {
    return locale in dictionary;
}
exports.hasLocaleDictionary = hasLocaleDictionary;
function getMessageFromDictionary(locale, id) {
    if (hasLocaleDictionary(locale)) {
        const localeDictionary = getLocaleDictionary(locale);
        if (id in localeDictionary) {
            return localeDictionary[id];
        }
        const ids = id.split('.');
        let tmpDict = localeDictionary;
        for (let i = 0; i < ids.length; i++) {
            if (typeof tmpDict[ids[i]] !== 'object') {
                return tmpDict[ids[i]] || null;
            }
            tmpDict = tmpDict[ids[i]];
        }
        return tmpDict;
    }
    return null;
}
exports.getMessageFromDictionary = getMessageFromDictionary;
function getClosestAvailableLocale(refLocale) {
    if (refLocale == null)
        return null;
    const relatedLocales = (0, utils_js_1.getPossibleLocales)(refLocale);
    for (let i = 0; i < relatedLocales.length; i++) {
        const locale = relatedLocales[i];
        if (hasLocaleDictionary(locale)) {
            return locale;
        }
    }
    return null;
}
exports.getClosestAvailableLocale = getClosestAvailableLocale;
function addMessages(locale, ...partials) {
    $dictionary.update(d => {
        d[locale] = Object.assign(d[locale] || {}, ...partials);
        return d;
    });
}
exports.addMessages = addMessages;
const $locales = /*@__PURE__*/ (0, store_1.derived)([$dictionary], ([$dictionary]) => Object.keys($dictionary));
exports.$locales = $locales;
$dictionary.subscribe(newDictionary => (dictionary = newDictionary));
