// precompile-intl-runtime/dist/commonjs/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.__time = exports.__date = exports.__number = exports.__select = exports.__plural = exports.__offsetPlural = exports.__interpolate = exports.getTimeFormatter = exports.getNumberFormatter = exports.getDateFormatter = exports.json = exports.time = exports.number = exports.date = exports.t = exports._ = exports.format = exports.formatMessage = exports.isLoading = exports.register = exports.addMessages = exports.locales = exports.dictionary = exports.locale = exports.init = exports.waitLocale = void 0;
const loaderQueue_js_1 = require("./includes/loaderQueue.js");
const utils_js_1 = require("./includes/utils.js");
__exportStar(require("./includes/localeGetters.js"), exports);
__exportStar(require("./includes/utils.js"), exports);
function waitLocale(locale) {
    return (0, loaderQueue_js_1.flush)(locale || (0, utils_js_1.getCurrentLocale)() || (0, utils_js_1.getOptions)().initialLocale);
}
exports.waitLocale = waitLocale;
var configs_js_1 = require("./configs.js");
Object.defineProperty(exports, "init", { enumerable: true, get: function () { return configs_js_1.init; } });
var locale_js_1 = require("./stores/locale.js");
Object.defineProperty(exports, "locale", { enumerable: true, get: function () { return locale_js_1.$locale; } });
var dictionary_js_1 = require("./stores/dictionary.js");
Object.defineProperty(exports, "dictionary", { enumerable: true, get: function () { return dictionary_js_1.$dictionary; } });
Object.defineProperty(exports, "locales", { enumerable: true, get: function () { return dictionary_js_1.$locales; } });
Object.defineProperty(exports, "addMessages", { enumerable: true, get: function () { return dictionary_js_1.addMessages; } });
var loaderQueue_js_2 = require("./includes/loaderQueue.js");
Object.defineProperty(exports, "register", { enumerable: true, get: function () { return loaderQueue_js_2.registerLocaleLoader; } });
var loading_js_1 = require("./stores/loading.js");
Object.defineProperty(exports, "isLoading", { enumerable: true, get: function () { return loading_js_1.$isLoading; } });
const formatters_js_1 = require("./stores/formatters.js");
var formatters_js_2 = require("./stores/formatters.js");
Object.defineProperty(exports, "formatMessage", { enumerable: true, get: function () { return formatters_js_2.formatMessage; } });
Object.defineProperty(exports, "format", { enumerable: true, get: function () { return formatters_js_2.$format; } });
Object.defineProperty(exports, "_", { enumerable: true, get: function () { return formatters_js_2.$format; } });
Object.defineProperty(exports, "t", { enumerable: true, get: function () { return formatters_js_2.$format; } });
Object.defineProperty(exports, "date", { enumerable: true, get: function () { return formatters_js_2.$formatDate; } });
Object.defineProperty(exports, "number", { enumerable: true, get: function () { return formatters_js_2.$formatNumber; } });
Object.defineProperty(exports, "time", { enumerable: true, get: function () { return formatters_js_2.$formatTime; } });
Object.defineProperty(exports, "json", { enumerable: true, get: function () { return formatters_js_2.$getJSON; } });
// low-level
var formatters_js_3 = require("./includes/formatters.js");
Object.defineProperty(exports, "getDateFormatter", { enumerable: true, get: function () { return formatters_js_3.getDateFormatter; } });
Object.defineProperty(exports, "getNumberFormatter", { enumerable: true, get: function () { return formatters_js_3.getNumberFormatter; } });
Object.defineProperty(exports, "getTimeFormatter", { enumerable: true, get: function () { return formatters_js_3.getTimeFormatter; } });
function __interpolate(value) {
    return value === 0 ? 0 : value || '';
}
exports.__interpolate = __interpolate;
const PLURAL_RULES = Object.create(null);
function getLocalPluralFor(v) {
    let loc = (0, utils_js_1.getCurrentLocale)();
    let pluralRules = PLURAL_RULES[loc] || (PLURAL_RULES[loc] = new Intl.PluralRules(loc));
    let key = pluralRules.select(v);
    return key === 'other' ? 'h' : key[0];
}
function __offsetPlural(value, offset, opts) {
    return opts[value] || opts[getLocalPluralFor(value - offset)] || "";
}
exports.__offsetPlural = __offsetPlural;
function __plural(value, opts) {
    return opts[value] || opts[getLocalPluralFor(value)] || "";
}
exports.__plural = __plural;
function __select(value, opts) {
    return opts[value] || opts['other'] || '';
}
exports.__select = __select;
function __number(value, format) {
    return (0, formatters_js_1.formatNumber)((0, utils_js_1.getCurrentLocale)(), value, typeof format === 'string' ? { format } : format);
}
exports.__number = __number;
function __date(value, format = "short") {
    return (0, formatters_js_1.formatDate)((0, utils_js_1.getCurrentLocale)(), value, { format });
}
exports.__date = __date;
function __time(value, format = "short") {
    return (0, formatters_js_1.formatTime)((0, utils_js_1.getCurrentLocale)(), value, { format });
}
exports.__time = __time;
